src/components/Card/favoritesCard.js

```diff
@@ -13,90 +13,80 @@ const IndexCalendar = dynamic(() => import('../Calendar/indexCalendar'), {
+	const workouts = useStore(state => state.workouts);
+	const currentExercises = workouts.filter(workout => exerciseCard.name === workout.name);
+	const sortedCurrentExercises = currentExercises.sort(
+		(a, b) => new Date(b.date) - new Date(a.date)
+	);
+	const latestStats = sortedCurrentExercises.length
+		? sortedCurrentExercises[0]
+		: {sets: '0', reps: '0', weight: '0'};
 
-	function handleWeightInput(input) {
-		changeWeight(card.id, Number(input));
+	function selectDate(selectedDate) {
+		const date = selectedDate.toDateString();
+		addToPlanner(
+			date,
+			exerciseCard.name,
+			latestStats.sets,
+			latestStats.reps,
+			latestStats.weight
+		);
+		setIsCalendarVisible(false);
 	}
 
 	return (
 		<StyledCard>
 			<StyledCardHeader>
-				<StyledCardHeadline>{card.name}</StyledCardHeadline>
-				<BookmarkButton
-					onClick={() => {
-						debookmark(card.id);
-						setIsBookmarked(!isBookmarked);
-					}}
-				>
-					{isBookmarked ? (
-						<FilledBookmarkIcon id={card.id} />
+				<StyledCardHeadline>{exerciseCard.name}</StyledCardHeadline>
+				<BookmarkButton onClick={() => handleBookmark(exerciseCard.id)}>
+					{exerciseCard.isBookmarked ? (
+						<FilledBookmarkIcon id={exerciseCard.id} />
 					) : (
-						<NotFilledBookmarkIcon id={card.id} />
+						<NotFilledBookmarkIcon id={exerciseCard.id} />
 					)}
 				</BookmarkButton>
 				<StyledCalendarButton
 					onClick={() => {
-						setIsShown(!isShown);
+						setIsCalendarVisible(!isCalendarVisible);
 					}}
 				>
 					add to planner
 				</StyledCalendarButton>
-				{isShown && <IndexCalendar id={card.id} />}
+				{isCalendarVisible && <IndexCalendar onSelectDate={selectDate} />}
 			</StyledCardHeader>
 			<StyledCardBody>
-				<Image src={card.image} alt="dummy fitness image" width={300} height={200} />
+				<Image
+					src={exerciseCard.image}
+					alt="dummy fitness image"
+					width={300}
+					height={200}
+				/>
 				<StyledHideButton
 					onClick={() => {
-						setShowText(!showText);
-						handleChange();
+						setIsInstructionVisible(!isInstructionVisible);
 					}}
 				>
-					{buttonText ? 'Show instruction' : 'Hide instruction'}
+					{isInstructionVisible ? 'Hide instruction' : 'Show instruction'}
 				</StyledHideButton>
-				{showText && (
-					<StyledCardDescription id={card.id}>{card.description}</StyledCardDescription>
+				{isInstructionVisible && (
+					<StyledCardDescription>{exerciseCard.instruction}</StyledCardDescription>
 				)}
 			</StyledCardBody>
-			<form onSubmit={handleSubmit}>
-				<CardFooter
-					sets={card.sets}
-					reps={card.reps}
-					weight={card.weight}
-					handleSetInput={handleSetInput}
-					handleRepInput={handleRepInput}
-					handleWeightInput={handleWeightInput}
-				/>
-			</form>
+			<StyledCardFooter>
+				Latest sets: {latestStats.sets} Latest reps: {latestStats.reps} Latest weight:{' '}
+				{latestStats.weight}
+			</StyledCardFooter>
```

Comment:
As you are not using `latestStats` anymore, I suggest removing it from your footer.